ui/app/controllers/workspace/projects/new.ts

```diff
@@ -0,0 +1,33 @@
+import Controller from '@ember/controller';
+import { inject as service } from '@ember/service';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import ApiService from 'waypoint/services/api';
+import { Project, UpsertProjectRequest } from 'waypoint-pb';
+import FlashMessagesService from 'waypoint/services/flash-messages';
+export default class WorkspaceProjectsNew extends Controller {
+  @service api!: ApiService;
+  @service flashMessages!: FlashMessagesService;
+
+  @tracked createGit = false;
+
+  @action
+  async saveProject() {
+    let project = this.model;
+    let ref = new Project();
```

Comment:
Isn't `model` here a project? Is there a reason you need a new `Project`? (really this is my unfamiliarity with grpc-web).

Reply:
so the model created in the route uses `Project.toObject()` and is therefore not a `Project` anymore, and there's no method to convert it back. Here we have to reconstruct the grpc-web object. Luckily in this case it is quite simple, but that process explains all the object gymnastics you saw in https://github.com/hashicorp/waypoint/pull/1057